### ISSUE TITLE
Missing files while training and testing. (SED only)

### ISSUE THREAD

**Issue by Bovey0809** [turpaultn/dcase20_task4#1]
Thanks for the great work.
I came across some problems while reproducing the baseline model result of SED.

1. when using python main.py or Testmodel.py, when running on this line.
`features = self.get_feature_file_func(self.feat_filenames.iloc[index])`, 
it will come across the "None" file error, which is caused by the missing files.

Is there any parameters I can modify to ignore the missing files?

Thanks. 

PS: The missing files list had been send to your mail.

**Comment by turpaultn** [comment]
Could you please send me the error you are receiving please ? (The best would be the complete log and you changing `terminal_level = logging.DEBUG`in config.py)

Because while extracting the dataframe with the features (`data_utils/Desed.py`) the missing files should be discarded.
I guess it is a problem there then.
Could you replace line 281-284 in Desed.py to:
```python
if not osp.isfile(wav_path):
    logger.error("File %s is in the tsv file but the feature is not extracted because "
                 "file do not exist!" % wav_path)
    out_path = None
    filename = None
```
This should do the trick I guess. 
If yes I could add it so the baseline is running even with some missing files.

**Comment by Bovey0809** [comment]
Thanks for replying. I have received the missing file link.(Haven't downloaded yet)
Here is the log of "python main.py".

[log.txt](https://github.com/turpaultn/dcase20_task4/files/4598446/log.txt)

And the env log is here.
[env_log.txt](https://github.com/turpaultn/dcase20_task4/files/4598460/env_log.txt)

After changing the code in Desed.py as you mentioned.
Another error comes out.
[afterChanging.txt](https://github.com/turpaultn/dcase20_task4/files/4598472/afterChanging.txt)

After finishing downloading the missing files, I will try it again.

Thanks.

**Comment by Bovey0809** [comment]
For someone who may come across the same error, try to modify the Desed.py around line 356-360.
```python
with closing(multiprocessing.Pool(n_jobs)) as p:
            for filename, out_path in tqdm(p.imap_unordered(extract_file_func, uniq_fpaths, 200), total=len(uniq_fpaths)):
                if not out_path:
                    df_meta.drop(df_meta[df_meta.filename == filename].index, inplace=True)
                    continue
                row_features = df_meta[df_meta.filename == filename]
                row_features.loc[:, "feature_filename"] = out_path
                df_features = df_features.append(
                    row_features, ignore_index=True)

**Comment by turpaultn** [comment]
Fixed by #3
